llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py

```diff
@@ -31,7 +32,24 @@ def __init__(
         self._node_label = node_label
 
         self._driver = FalkorDB.from_url(url).select_graph(database)
-        self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
+
+        try:
+            result_set = self._driver.query(
+                f"""
+                CYPHER lbl='{self._node_label}' prop='id'
+                CALL db.indexes() YIELD label, properties
+                WHERE label = $lbl AND $prop in properties
+                RETURN 1 AS index_exists
+                """
+            ).result_set
+            # TODO: this check may cause a race issue due to these two queries are not atomic.
```

Comment:
Hmm I think maybe what we had before is better than potentially introducing another error/fault.

Reply:
@nerdai agree, I prefer the previous solution, @logan-markewich  what do you think? revert or keep this?

Reply:
Reading from his last message I'd infer that he would want to revert to previous solution as well.

Reply:
@nerdai reverted!